pkg/types/gcp/validation/machinepool.go

```diff
@@ -80,3 +84,45 @@ func ValidateDefaultDiskType(p *gcp.MachinePool, fldPath *field.Path) field.Erro
 
 	return allErrs
 }
+
+func ValidateKMSKey(platform *gcp.Platform, p *types.MachinePool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if p != nil && p.Platform.GCP != nil && p.Platform.GCP.OSDisk.EncryptionKey != nil {
+		kmsClient, err := kms.NewKeyManagementClient(context.Background())
+		if err != nil {
+			return append(allErrs, field.InternalError(fldPath.Child("encryptionKey"), err))
+		}
+		defer kmsClient.Close()
+
+		project := p.Platform.GCP.OSDisk.EncryptionKey.KMSKey.ProjectID
+		if project == "" {
+			project = platform.ProjectID
+		}
+		location := p.Platform.GCP.OSDisk.EncryptionKey.KMSKey.Location
+		if location == "" {
+			location = platform.Region
+		}
+
+		req := &kmspb.ListKeyRingsRequest{
+			Parent: fmt.Sprintf("projects/%s/locations/%s", project, location),
+		}
+
+		it := kmsClient.ListKeyRings(context.Background(), req)
```

Comment:
Can you use [GetKeyRing](https://pkg.go.dev/cloud.google.com/go/kms/apiv1#KeyManagementClient.GetKeyRing) to avoid the iterator and simplify this?